src/Features/LanguageServer/Protocol/Handler/CodeActions/RoslynNestedCodeAction.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Runtime.Serialization;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActions;
+
+[DataContract]
+internal sealed class RoslynNestedCodeAction(ImmutableArray<RoslynNestedCodeAction> nestedActions) : CodeAction
```

Comment:
this name is super strange.  it seems to indicate that it's a nested code action.  but really it's a code action *with* nested actions.